File: client/go/loggingclient/LogOnOffManager.go
package loggingclient

import (
	"io/ioutil"
	"net/http"
	"strconv"
	"strings"
	"time"
)

// LogOnOffManager 日志开关管理类
type LogOnOffManager struct {
	LogOnOffCacheTimeOut int

	LogOnOffGetURL string

	LastUpdateTime time.Time

	LogOnOff *LogOnOff
}

// NewLogOnOffManager 实例化一个LogOnOffManager引用对象
func NewLogOnOffManager() *LogOnOffManager {

	var loom = new(LogOnOffManager)
	loom.LogOnOffCacheTimeOut = LoggingSettings.LogOnOffCacheTimeOut
	loom.LogOnOffGetURL = LoggingSettings.LoggingServerHost + "/GetLogOnOff.ashx?appId=" + strconv.Itoa(int(LoggingSettings.AppId))
	loom.LogOnOff = &LogOnOff{Debug: 1, Error: 1, Info: 1, Warm: 1}
	loom.LastUpdateTime = time.Now()
	return loom
}

// GetLogOnOff 获取日志开关
func (loom *LogOnOffManager) getLogOnOff() (ret *LogOnOff) {
	if &loom.LogOnOff == nil {
		return &LogOnOff{Debug: 1, Error: 1, Info: 1, Warm: 1}
	}
	return loom.LogOnOff
}

// RefreshLogOnOff 从服务端获取并刷新日志开关,10分钟缓存
func (loom *LogOnOffManager) RefreshLogOnOff() {

	var timeDiff = time.Now().Sub(loom.LastUpdateTime).Minutes()
	if int(timeDiff) < loom.LogOnOffCacheTimeOut {
		return
	}
	response, _ := http.Get(loom.LogOnOffGetURL)

	body, _ := ioutil.ReadAll(response.Body)
	var resp = string(body)
	if resp != "" {
		var arr = strings.Split(resp, ",")

		var _debug, _ = strconv.Atoi(arr[0])
		var _info, _ = strconv.Atoi(arr[1])
		var _warm, _ = strconv.Atoi(arr[2])
		var _error, _ = strconv.Atoi(arr[3])

		loom.LogOnOff.Debug = byte(_debug)
		loom.LogOnOff.Info = byte(_info)
		loom.LogOnOff.Warm = byte(_warm)
		loom.LogOnOff.Error = byte(_error)

	}
	loom.LastUpdateTime = time.Now()
	defer response.Body.Close()
}

File: client/dotnetcore/Logging.Client/LogOnOffManager.cs
using System;

namespace Logging.Client
{
    internal sealed class LogOnOffManager
    {
        private LogOnOffManager()
        {
        }

        private readonly static int LogOnOffCacheTimeOut = 10;//单位:分钟

        private readonly static string GetLogOnOffUrl = Settings.LoggingServerHost + "/GetLogOnOff.ashx?appId=" + Settings.AppId;

        private readonly static LogOnOff Default = new LogOnOff { Debug = 1, Error = 1, Info = 1, Warm = 1 };

        private static DateTime LastUpdateTime;

        private static LogOnOff logOnOff = null;

        public static LogOnOff GetLogOnOff()
        {
            if (logOnOff == null) { return Default; }
            return logOnOff;
        }

        /// <summary>
        /// 从服务端获取并刷新日志开关,10分钟缓存
        /// </summary>
        /// <returns></returns>
        public static void RefreshLogOnOff()
        {
            if ((DateTime.Now - LastUpdateTime).TotalMinutes < LogOnOffCacheTimeOut) { return; }

            string resp = string.Empty;

            try
            {
                //using (WebClientEx _client = new WebClientEx(10 * 1000))
                //{
                //    byte[] resp_byte = _client.DownloadData(GetLogOnOffUrl);
                //    resp = Encoding.UTF8.GetString(resp_byte);
                //}
                WebClientEx _client = new WebClientEx();
                resp = _client.GetWebRequest(new Uri(GetLogOnOffUrl));
            }
            catch { }
            if (!string.IsNullOrWhiteSpace(resp))
            {
                logOnOff = new LogOnOff();
                string[] arr = resp.Split(',');
                logOnOff.Debug = Convert.ToByte(arr[0]);
                logOnOff.Info = Convert.ToByte(arr[1]);
                logOnOff.Warm = Convert.ToByte(arr[2]);
                logOnOff.Error = Convert.ToByte(arr[3]);
            }
            LastUpdateTime = DateTime.Now;
        }
    }
}